src/App.tsx

```diff
@@ -1,24 +1,84 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { getTodos } from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { GlobalLoader } from './types/GlobalLoader';
+
+import { TodoList } from './components/TodoList';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoError } from './components/TodoError';
+
+import { USER_ID } from './utils/userId';
+import { TodoContext } from './components/TodoContext';
+import { ErrorContext } from './components/ErrorContext';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const { todos, setTodos } = useContext(TodoContext);
+  const { setError } = useContext(ErrorContext);
+  const [status, setStatus] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [
+    globalLoader,
+    setGlobalLoader,
+  ] = useState<GlobalLoader>(GlobalLoader.Non);
+
+  const filtredTodos = useMemo(() => todos.filter(({ completed }) => {
+    switch (status) {
+      case Status.Active:
+        return !completed;
+      case Status.Completed:
+        return completed;
+      default:
+        return true;
+    }
+  }), [status, todos]);
+
+  useEffect(() => {
+    setError('');
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to download todos');
+      });
+  }, []);
```

Comment:
Also you can move ur function outside of component and just pass it into useMemo